ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/moreh_clip_grad_norm_step1_program_factory.cpp

```diff
@@ -0,0 +1,222 @@
+    const uint32_t im5_t = 1;  // |x|^p * exp(log(|x|) * decimal)
+
+    const auto cb_data_format = tt::tt_metal::datatype_to_dataformat_converter(tmp_pow_sum.get_dtype());
+
+    ttnn::operations::CreateCircularBuffer(
+        program,
+        core_group_1,
+        cb_data_format,
+        {
+            {tt::CB::c_in0, in0_t},        // input(==x)
+            {tt::CB::c_in1, in1_t},        // one
+            {tt::CB::c_in2, in2_t},        // decimal
+            {tt::CB::c_in3, in3_t},        // mask_h_w
+            {tt::CB::c_out0, out0_t},      // output(==y)
+            {tt::CB::c_intermed0, im0_t},  // |x|
+            {tt::CB::c_intermed1, im1_t},  // |x|^p
+            {tt::CB::c_intermed2, im2_t},  // Add[|x|^p * exp(log(|x|) * decimal)]
+            {tt::CB::c_intermed3, im3_t},  // log(|x|)
+            {tt::CB::c_intermed4, im4_t},  // exp(log(|x|) * decimal)
+            {tt::CB::c_intermed5, im5_t},  // |x|^p * exp(log(|x|) * decimal)
+        });
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      DataMovementKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    const auto reader_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "reader_moreh_clip_grad_norm_step1.cpp";
+    const auto writer_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "writer_moreh_clip_grad_norm_step1.cpp";
+
+    const auto reader_kernel_id = ttnn::operations::CreateReadKernel(program, reader_kernel_file, core_group_1);
+    const auto writer_kernel_id = ttnn::operations::CreateWriteKernel(program, writer_kernel_file, core_group_1);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      ComputeKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    std::map<std::string, std::string> compute_defines{};
+    compute_defines["REDUCE_OP"] = "PoolType::SUM";
+    compute_defines["REDUCE_DIM"] = "ReduceDim::REDUCE_SCALAR";
+
+    const auto compute_kernel_file =
+        "ttnn/cpp/ttnn/operations/moreh/moreh_clip_grad_norm/moreh_clip_grad_norm_step1/device/kernels/"
+        "moreh_clip_grad_norm_step1_kernel.cpp";
+
+    const auto compute_kernel_id = ttnn::operations::CreateComputeKernel(
+        program, compute_kernel_file, {core_group_1, num_inputs_per_core_group_1}, compute_defines);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      RuntimeArgs SetUp
+    ////////////////////////////////////////////////////////////////////////////
+
+    const auto output_addr = tmp_pow_sum.buffer()->address();
+    bool tmp_pow_sum_is_dram = tmp_pow_sum.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
```

Comment:
You can use `is_dram()` here instead of the ternary operation

Reply:
Resolved in [367b454](https://github.com/tenstorrent/tt-metal/pull/14667/commits/367b454f49e71941dc71e9cf323bf305bcb799ba)